bpf/lib/common.h

```diff
@@ -147,6 +147,12 @@ __revalidate_data(struct __ctx_buff *ctx, void **data_, void **data_end_,
 #define revalidate_data_first(ctx, data, data_end, ip)			\
 	__revalidate_data(ctx, data, data_end, (void **)ip, sizeof(**ip), true)
 
+/* revalidate_data_with_pull() does the same as revalidate_data_first(), except
+ * that the skb pulling is controlled with the "pull" parameter.
+ */
+#define revalidate_data_with_pull(ctx, data, data_end, ip, pull)	\
```

Comment:
Happy to change the function name with a better one.

Reply:
Yeah agree, the `revalidate_data_first()` vs `revalidate_data_with_pull()` is a bit hard to differ based on name.
Maybe rename `revalidate_data_first()` into `revalidate_data_pull()` which internally calls `__revalidate_data_pull(..., true)` such that the two occurences with `!from_host` as param use `__revalidate_data_pull(..., !from_host)`. Other than that lgtm, nice!